torch/distributed/fsdp/_init_utils.py

```diff
@@ -585,18 +591,40 @@ def _get_ignored_modules(
 def _get_ignored_params(
     root_module: torch.nn.Module,
     ignored_modules: Set[torch.nn.Module],
+    ignored_parameters: Optional[Iterable[torch.nn.Parameter]] = None,
 ) -> Set[torch.nn.Parameter]:
     """
-    Returns the parameters of the modules in ``ignored_modules``,
-    excluding any :class:`FlatParameter` s.
+    Returns the parameters of the modules in ``ignored_modules`` and
+    the parameters in ``ignored_parameters``, excluding any :class:`FlatParameter` s.
     """
-    return set(
+    all_ignored_params: Set[torch.nn.Parameter] = set()
+
+    params_in_ignored_modules = set(
         p
         for m in ignored_modules
         for p in m.parameters()
         if not _is_fsdp_flattened(p)
     )
 
+    all_ignored_params.update(params_in_ignored_modules)
+
+    if ignored_parameters is not None:
+        params_in_ignored_parameters = set(
+            p
+            for p in ignored_parameters
+            if not _is_fsdp_flattened(p)
+        )
+        all_ignored_params.update(params_in_ignored_parameters)
```

Comment:
Before this `update()`, what should be in `all_ignored_params`?

IIUC, this should only contain parameters from composable APIs that are incompatible with `fully_shard` (e.g. `replicate`) because those are the only ignored modules manually added into the `ignored_modules` set, not specified by the user. This means that for the wrapper FSDP path, `all_ignored_params` should be empty.

I want to double check if that is your understanding as well.

Reply:
That is also my understanding based on current code reading